Projects/My_website/myWebsite/blog/views.py

```diff
@@ -18,4 +18,45 @@ def postDetail(request,slug):
     return render(request,"blog/postDetail.html",{'post':iPost,"post_tag":iPost.tags.all()})
 
 def createBlog(request):
-    return render(request,'blog/createBlog.html')
\ No newline at end of file
+    if request.method== 'POST':
+        title = request.POST['title']
+        excerpt = request.POST['excerpt']
+        content = request.POST['content']
+        tag = request.POST['tag']
+        image = request.POST['file']
+
+       
+        if request.user.is_authenticated:
+
+            first_name = request.user.first_name
+            last_name = request.user.last_name
+            email = request.user.email
+            author=None
+
+            if Author.objects.filter(email=email).exists():
+                author = Author.objects.filter(email=email).first()
+            else:
+                first_name = request.user.first_name
+                last_name = request.user.last_name
+                email = request.user.email
+                auth = Author(first_name = first_name,last_name=last_name,email = email)
+                auth.save()
+                author = Author.objects.filter(email=email).first()
```

Comment:
@codal-uchoudhary , same thing here. at line 42, you're creating an author instance which can be directly used in creating the Post instance. why query again at 44 ?